app/controllers/tracks/approaches_controller.rb

```diff
@@ -52,28 +52,21 @@ def introduction
     {
       html: Markdown::Parse.(@exercise.approaches_introduction),
       avatar_urls: introduction_avatar_urls,
-      num_authors: introduction_num_authors,
-      num_contributors: introduction_num_contributors,
+      num_authors: @exercise.approach_introduction_authors.count,
+      num_contributors: @exercise.approach_introduction_contributors.count,
       updated_at: @exercise.approaches_introduction_last_modified_at,
       links: {
-        new: "https://github.com/exercism/#{@track.slug}/new/main/exercises/#{@exercise.git_type}/#{@exercise.slug}/.approaches/introduction.md?filename=introduction.md",
-        edit: "https://github.com/exercism/#{@track.slug}/edit/main/exercises/#{@exercise.git_type}/#{@exercise.slug}/.approaches/introduction.md"
+        edit: @exercise.approaches_introduction_edit_url
       }
     }
   end
 
-  memoize
-  def introduction_num_authors = @exercise.approach_introduction_authors.count
-
-  memoize
-  def introduction_num_contributors = @exercise.approach_introduction_contributors.count
-
   def introduction_avatar_urls
     avatar_urls = proc { |users, limit| users.order("RAND()").limit(limit).select(:avatar_url).to_a.map(&:avatar_url) }
 
     target = 3
     urls = avatar_urls.(@exercise.approach_introduction_authors, target)
-    if urls.size < 3 && introduction_num_contributors.positive?
+    if urls.size < 3 && @exercise.approach_introduction_contributors.count.positive?
```

Comment:
```suggestion
    if urls.size < 3 && @exercise.approach_introduction_contributors.length.positive?
```

If you're materialising them, it's better to use `length` rather than `count` as `length` will use the materialized array whereas `count` will do a query.